src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -1802,7 +1805,6 @@ class spark_resource_adaptor final : public rmm::mr::device_memory_resource {
             if (is_for_cpu == t_state.is_cpu_alloc) {
               transition(t_state, thread_state::THREAD_ALLOC_FREE);
             }
-            if (!is_for_cpu) { t_state.gpu_memory_allocated_bytes -= num_bytes; }
```

Comment:
I didn't notice in the original change that this code path is only running on *other* threads which is incorrect for the tracking behavior (see line 1787)